lambda/postSchedulePeerTeachers/Scheduler/src/main/java/scheduler/Handler.java

```diff
@@ -304,33 +307,49 @@ static void schedulePeopleToSections() {
             HashSet<Integer> badSections = new HashSet<>();
 
             // Schedule what you prefer or marked indifferent
-            for (Preference preference : frontPerson.getPreferences()) {
+            for (Preference preference : frontPerson.getSortedPreferences()) {
                 if (frontPerson.getCurrentAssignments() >= frontPerson.getDesiredNumberAssignments()) break;
                 if (!possibleSections.contains(preference.getSectionId())) continue;
                 if (preference.isPreferable()) {
                     Section section = sections.get(preference.getSectionId());
                     if (section.getAssignedPTs() < section.getCapacityPTs()) {
-                        section.addAssignedPTs();
-                        frontPerson.addCurrentAssignment();
-                        schedule.putIfAbsent(preference.getSectionId(), new ArrayList<>());
-                        schedule.get(preference.getSectionId()).add(frontPerson.getPersonId());
+                        boolean alreadyAssigned = false;
+                        for (Meeting meeting : section.getMeetings()) {
+                            if (frontPerson.alreadyAssigned(meeting.getWeekday(), meeting.getStartTime(), meeting.getEndTime())) {
+                                alreadyAssigned = true;
+                            }
+                        }
+                        if (!alreadyAssigned) {
+                            section.addAssignedPTs();
+                            frontPerson.addCurrentAssignment(section);
+                            schedule.putIfAbsent(preference.getSectionId(), new ArrayList<>());
+                            schedule.get(preference.getSectionId()).add(frontPerson.getPersonId());
+                        }
```

Comment:
This is repeated a few lines down- can it be stuck in a lambda? Does Java have lambdas? Anon functions? Idk